pandas/core/arrays/datetimelike.py

```diff
@@ -1679,3 +1679,27 @@ def _ensure_datetimelike_to_i8(other, to_utc=False):
             # period array cannot be coerced to int
             other = Index(other)
     return other.asi8
+
+
+def is_valid_na(obj, dtype):
```

Comment:
I wouldn't put this here, at the very least should be in pandas.core.dtypes.missing (if you can't group in cython).

also name is not very descriptive. how is_null_datetimelike_for_compatible_dtypes (I know its long but then its descriptive).

Reply:
will do